src/python/pim/polynomial_representation.py

```diff
@@ -0,0 +1,89 @@
+from typing import List
+import re
+import itertools as it
+import numpy as np
+
+class Polynomial:
+    def __init__(self, coefficients: List[int]) -> None:
+        self.coefficients: List[int] = coefficients
+
+    def __str__(self) -> str:
+        """returns a string representation of the polynomial"""
+        coeff_list: List[str] = [display(c, i) for i, c in enumerate(self.coefficients)]
+        filter_coeff_list = list(filter(lambda x: x != "", coeff_list))
+        return re.sub(r"\+ -", "- ", " + ".join(filter_coeff_list))
+
+    def __add__(self, other: "Polynomial") -> "Polynomial":
+        """returns the sum of two polynomials (like degrees are added together)"""
+        return [sum(n) for n in it.zip_longest(self.coefficients, other.coefficients, fillvalue = 0)]
+
+    def __mul__(self, other: "Polynomial") -> "Polynomial":
+        """returns the product of two polynomials"""
+        len_self: int = len(self.coefficients)
+        len_other: int = len(other.coefficients)
+        blank_coeff_list: list(int) = [0] * (len_self + len_other - 1)
+
+        #Multiplies two polynomials term by term
+        for i in range(len_self):
+            for j in range(len_other):
+                blank_coeff_list[i + j] += self.coefficients[i] * other.coefficients[j]
```

Comment:
Suppose we have two instances of `Polynomial` `p1` and `p2`:

```python 
p1 : Polynomial = Polynomial([1])
p2 : Polynomial = Polynomial([1])
```

Then `p1 * p2` should be something that represents `x**2`, which would be `Polynomial([0, 1])`. It seems like the output value `blank_coeff_list` would have been `[1]` with this implementation. Is that correct?

Relatedly, this function returns a value of type `List[int]`. Just like above, is that desired? Or should it be of type `Polynomial`?

Reply:
I'm fixing the first half of your question, but for the second half I'm slightly confused. Because the way we enter a Polynomial initially is a List[int]. If I did want to return a `Polynomial` is that simply me wrapping the output in `Polynomial`? Why would or wouldn't I want to do that?
`g: Polynomial = Polynomial([-8, 17, 0, 5])`

Reply:
It is just wrapping like you have there. You would want to do that because the behavior on an instance of `Polynomial` is different than the behavior of the list. Also, just conceptually, if you multiply two polynomials together, shouldn't you get a polynomial back?

Reply:
Ok I get that - thanks! (also I should refresh the page because I don't see your comments and then just keep doing Sarah things haha)